### ISSUE TITLE
SilverStripe 3 class names containing a _ conflict with tables that contain a \

### ISSUE THREAD

**Issue by Zauberfisch** [silverstripe/silverstripe-framework#7431]
If a table named `foo\bar\blub` exists in the database and a class named 
`foo_bar_blub`, SilverStripe will fail to run /dev/build with the following error message:

    [User Error] Uncaught SS_DatabaseException: Couldn't run query: SHOW FULL FIELDS IN "foo_bar_blub" Table 'silverstripe.foo_bar_blub' doesn't exist

my guess is that the "does the table exist" check somehow translates _ to \, but other parts of the database abstraction don't.
This happened to me because I created a namespaced class and renamed the class to a non-namespaced class with the former namespace as prefix and ran a /dev/build.

**Comment by chillu** [comment]
> This happened to me because I created a namespaced class and renamed the class to a non-namespaced class with the former namespace as prefix and ran a /dev/build.

I find it hard to follow this bug report. Was this in the process of a 3 -> 4 migration? Can you describe this step-by-step please?

**Comment by Zauberfisch** [comment]
No, this affects 3, I have not tested 4 (though I see potential that this would also affect SilverStripe 4).

I was back porting a namespaced class to SilverStripe 3.
I copied that class with namespaces into the SilverStripe 3 project, ran /dev/build which created the namespaced classes. Only after that I renamed the class to not contain namespaces and ran /dev/build again upon which I received said error.

I am not certain if it's worth fixing this. But because this issue does not throw any meaningful error message, I see how it could be hard to debug for someone who ran into it.
